vllm/model_executor/layers/quantization/modelopt.py

```diff
@@ -1650,16 +1651,15 @@ def apply(
             use_llama4_routing = (
                 custom_routing_function is Llama4MoE.custom_routing_function
             )
-            routing_method_type = flashinfer.RoutingMethodType.DeepSeekV3
+            routing_method_type = layer.routing_method_type
             if use_llama4_routing:
-                routing_method_type = flashinfer.RoutingMethodType.Llama4
+                routing_method_type = RoutingMethodType.Llama4
+            router_logits = router_logits.to(torch.float32) if routing_method_type == RoutingMethodType.DeepSeekV3 else router_logits
```

Comment:
should we leave instead leave this abstraction to be handled inside flashinfer implementation of trtllm_fp4_block_scale_moe and potentially trtllm_fp8_block_scale_moe too?